### PR TITLE
feat: support png and jpeg files for connector logo

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Support png and jpeg file for connector logo.

## Testing

Tested locally with png and jpeg connector logo files. Logos, config templates and README files can display properly.
image
image
image

### CODE REVIEW

File: packages/core/src/connectors/utilities/index.ts [hunk 1]
```diff
@@ -34,27 +34,61 @@ export function validateConnectorModule(
...
 }
 
-export const readUrl = async (
-  url: string,
-  baseUrl: string,
-  type: 'text' | 'svg'
-): Promise<string> => {
+type UrlType = 'text' | 'svg' | 'png' | 'jpeg' | undefined;
+
+const getUrlType = (url: string): UrlType => {
```
Comment (wangsijie) [inline packages/core/src/connectors/utilities/index.ts:39]: Maybe regex can make this funciton more readable

File: packages/core/src/connectors/utilities/index.ts [hunk 2]
```diff
@@ -34,27 +34,61 @@ export function validateConnectorModule(
   }
 }
 
-export const readUrl = async (
-  url: string,
-  baseUrl: string,
-  type: 'text' | 'svg'
-): Promise<string> => {
+type UrlType = 'text' | 'svg' | 'png' | 'jpeg' | undefined;
```
Comment (charIeszhao) [inline packages/core/src/connectors/utilities/index.ts:37]: Don't forget `jpg`

File: packages/core/src/connectors/utilities/index.ts [hunk 3]
```diff
@@ -34,27 +34,61 @@ export function validateConnectorModule(
...
+const getUrlType = (url: string): UrlType => {
+  const splitUrl = url.split('/').filter(Boolean);
+
+  assertThat(
+    splitUrl.length > 1,
+    new ConnectorError(ConnectorErrorCodes.InvalidMetadata, { message: `Invalid url: ${url}.` })
+  );
+
+  const splitBaseUrl = splitUrl.slice(-1)[0]?.split('.').filter(Boolean);
+  const extension = splitBaseUrl && splitBaseUrl.length > 1 ? splitBaseUrl.slice(-1)[0] : undefined;
```
Comment (charIeszhao) [inline packages/core/src/connectors/utilities/index.ts:48]: If we only wants to get the extension, we can use `.lastIndexOf('.')`

**Review by simeng-li — CHANGES_REQUESTED** [review]
Let's add ut for those utility methods.

**Pull request closed without merging**